### PR TITLE
feat: add helper methods for component disabling

### PR DESCRIPTION
**Pull request by sillyElvy**
## About

This pull request adds methods to disable components

## Checklist

- [x] I've ran `pre-commit` to format and lint the change(s) made.
- [ ] I've checked to make sure the change(s) work on `3.8.6` and higher.
- [ ] This fixes/solves an [Issue](https://github.com/goverfl0w/discord-interactions/issues) (If existent):.
  - resolves #
- I've made this pull request for/as: (check all that apply)
  - [ ] Documentation
  - [ ] Breaking change
  - [x] New feature/enhancement
  - [ ] Bugfix

### CODE REVIEW

File: interactions/client/context.py [hunk 1]
```diff
@@ -654,6 +654,37 @@ async def defer(
...
+                for component in components.components:
+                    component.disabled = True
+
+            if other_kwargs.get("components"):
+                raise LibraryException(
+                    12, "You must not specify the `components` argument in this method."
+                )
+
+            other_kwargs["components"] = self.message.components
+            await self.edit(**other_kwargs)
```
Comment (Catalyst4222) [inline interactions/client/context.py]: You have the return type as a Message, but if the user chooses to respond to the interaction, nothing gets returned

Comment (sillyElvy) [reply interactions/client/context.py]: fixed

**Review by Toricane — COMMENTED** [review]
Could you also add a utility function that takes in a list of action rows or components and disables them all?
I think that would be useful as well

**Comment by sillyElvy** [comment]
> Could you also add a utility function that takes in a list of action rows or components and disables them all? I think that would be useful as well
Done

**Review by Catalyst4222 — APPROVED** [review]

**Pull request merged**